crates/bevy_reflect/src/type_registry.rs

```diff
@@ -206,14 +221,15 @@ impl TypeRegistration {
     }
 
     /// Creates type registration information for `T`.
-    pub fn of<T: Reflect>() -> Self {
+    pub fn of<T: Reflect + Typed>() -> Self {
```

Comment:
I went ahead and added this restriction. The alternative is to change the `type_info: TypeInfo` to `type_info: Option<TypeInfo>` and set it during registration (like we do with `TypeData`).

Reply:
Any argument as to why we shouldn't just merge `Typed` into `Reflect` or add it as a `Reflect` bound?

Reply:
Two reasons:

1. It removes the `Self: Sized` bound on the `type_info()` method since we need `Reflect` to be object-safe. That's also why we can't add it as a bound to `Reflect` unfortunately.
2. If we ever wanted to make just `Typed` derivable/implementable without a requiring `Reflect` (for whatever reason), we could do that

But point 2 probably doesn't matter and I'm open to moving it back to `Reflect` if we're okay with the `Sized` bound.

Reply:
>  I'm open to moving it back to Reflect if we're okay with the Sized bound.

I don't think that'll work: we need to be able to reflect container types too. Perhaps a doc comment on `Typed`?

Reply:
> Perhaps a doc comment on `Typed`?

Stating the reason this isn't on `Reflect`?

Reply:
> I don't think that'll work: we need to be able to reflect container types too. Perhaps a doc comment on `Typed`?

I thought `?Sized` was reserved for trait objects and embedded slice types. `Vec<T>` and the like are sized regardless of `T`.